packages/atomic/src/utils/initialization-utils.tsx

```diff
@@ -122,36 +123,36 @@ export function InitializeBindings() {
     }
 
     component.componentWillLoad = function () {
-      const element = getElement(this);
-      const event = buildCustomEvent(
-        initializeEventName,
-        (bindings: Bindings) => {
-          this.bindings = bindings;
-
-          const updateLanguage = () => forceUpdate(this);
-          this.bindings.i18n.on('languageChanged', updateLanguage);
-          unsubscribeLanguage = () =>
-            this.bindings.i18n.off('languageChanged', updateLanguage);
-
-          try {
-            // When no controller is initialized, updating a property with a State() decorator, there will be no re-render.
-            // In this case, we have to manually trigger it.
-            this.initialize ? this.initialize() : forceUpdate(this);
-          } catch (e) {
-            this.error = e as Error;
+      return new Promise((resolve) => {
+        const element = getElement(this);
+        const event = buildCustomEvent(
+          initializeEventName,
+          async (bindings: Bindings) => {
+            this.bindings = bindings;
+
+            const updateLanguage = () => forceUpdate(this);
+            this.bindings.i18n.on('languageChanged', updateLanguage);
+            unsubscribeLanguage = () =>
+              this.bindings.i18n.off('languageChanged', updateLanguage);
+
+            try {
+              // When no controller is initialized, updating a property with a State() decorator, there will be no re-render.
+              // In this case, we have to manually trigger it.
+              await (this.initialize ? this.initialize() : forceUpdate(this));
+              resolve(componentWillLoad && componentWillLoad.call(this));
```

Comment:
I wouldn't pass anything as a parameter to `resolve`. Do you mean to `await` componentWillLoad?
```suggestion
              componentWillLoad && await componentWillLoad.call(this);
              resolve();
```

Reply:
actually, I also think that `resolve` should be moved outside of the `try catch`, just below it. Even when we error, we want to resolve.